source/extensions/transport_sockets/tls/ssl_socket.cc

```diff
@@ -223,13 +225,16 @@ void SslSocket::drainErrorQueue() {
     if (failure_reason_.empty()) {
       failure_reason_ = "TLS error:";
     }
-    absl::StrAppend(&failure_reason_, " ", err, ":",
+    absl::StrAppend(&failure_reason_, new_ssl_failure_format ? "|" : " ", err, ":",
                     absl::NullSafeStringView(ERR_lib_error_string(err)), ":",
                     absl::NullSafeStringView(ERR_func_error_string(err)), ":",
                     absl::NullSafeStringView(ERR_reason_error_string(err)));
   }
 
   if (!failure_reason_.empty()) {
+    if (new_ssl_failure_format) {
+      absl::StrAppend(&failure_reason_, ":TLS_error_end");
```

Comment:
I am not sure if `:TLS_error_end` is needed.  The last segment of the failure reason should be able to be identified without this or just with a simple delimiter (I don't have strong preference though)

Or actually do you just care about the whole error message between `TLS error:` and `:TLS_error_end`?Then do we have the needs for such individual segment that is separated by '|' delimiter.

Reply:
This string can be appended to other connection errors at connection level, and we want to parse since it is kind of too verbose.

Reply:
Ok. so you want to customize/simplified TLS error here with particular segments subtracted. I think overall approach make sense.
(1) we need to get the whole message between `TLS error:` and `:TLS_error_end`.  Those two tags should be unique. 
(2) we use `|` to subtract individual segments that we want.

If we can come up the unique end tag that is simpler, i think our search&parse can be faster.  I don't feel strongly though.  But we probably want to have their format matched e.g. TLS_error and TLS_error_end.

Reply:
Sure, we can use TLS_error.